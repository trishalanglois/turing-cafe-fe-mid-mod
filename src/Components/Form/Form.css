.resy-form {
  background-color: white;
  border: 2px solid black;
  width: 80%;
}

.form-button {
  margin: 1em;
}
