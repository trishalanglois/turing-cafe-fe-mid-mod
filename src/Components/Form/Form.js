import React, { Component } from 'react';
import './Form.css';

class Form extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      date: '',
      time: '',
      number: null
    }
  }

  render() {
    return (
      <>

        <h2>Make a reservation</h2>
        <p>Name</p>
        <input type="text" name="name" value={this.state.name} onChange={this.handleChange} />

        <p>Date</p>
        <input type="text" name="date" value={this.state.date} placeholder="MM/DD" onChange={this.handleChange} />

        <p>Time</p>
        <input type="text" name="time" value={this.state.time} placeholder="HH:MM" onChange={this.handleChange} />

        <p>Number of Guests</p>
        <input type="number" name="number" value={this.state.number} onChange={this.handleChange} />

      </>
    )
  }
}


export default Form;
