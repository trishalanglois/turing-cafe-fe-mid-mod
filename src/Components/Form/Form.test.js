import Form from './Form';
import React from 'react';
import { shallow } from 'enzyme';

describe('Form', () => {
  it('should match the snapshot with all data passed correctly', () => {
    const wrapper = shallow(<Form
      submitReservation={jest.fn()}
    />);

    expect(wrapper).toMatchSnapshot();
  })

  it('should update state when clearInputs is called', () => {
    const wrapper = shallow(<Form />);
    const mockState = {
      user: 'Trisha',
      date: '10/10',
      time: '10:10',
    };
    const mockClearedState = {
      user: '',
      date: '',
      time: '',
    };

    wrapper.instance().setState({ user: 'Trisha', date: '10/10', time: '10:10', number: 10})
    expect(wrapper.state('user')).toEqual('Trisha');
    expect(wrapper.state('date')).toEqual('10/10');
    expect(wrapper.state('time')).toEqual('10:10');
  })

  it('should call the submitReservation with the newReservation when the button is clicked', () => {
    const mockClick = jest.fn();
    const wrapper = shallow(<Form
      submitReservation={mockClick}
    />)

    wrapper.find('button').simulate('click');
    expect(mockClick).toHaveBeenCalled()
  })

})
