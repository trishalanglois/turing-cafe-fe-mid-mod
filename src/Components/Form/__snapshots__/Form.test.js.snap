// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the snapshot with all data passed correctly 1`] = `
<div
  className="reservation-form"
>
  <h2>
    Make a reservation
  </h2>
  <p>
    Name
  </p>
  <input
    name="name"
    onChange={[Function]}
    type="text"
    value=""
  />
  <p>
    Date
  </p>
  <input
    name="date"
    onChange={[Function]}
    placeholder="MM/DD"
    type="text"
    value=""
  />
  <p>
    Time
  </p>
  <input
    name="time"
    onChange={[Function]}
    placeholder="HH:MM"
    type="text"
    value=""
  />
  <p>
    Number of Guests
  </p>
  <input
    name="number"
    onChange={[Function]}
    type="number"
    value={null}
  />
  <br />
  <button
    className="form-button"
    onClick={[Function]}
  >
    Submit
  </button>
</div>
`;
